evm_arithmetization/src/poseidon/poseidon_stark.rs

```diff
@@ -28,20 +30,157 @@ use super::columns::{
+        *Table::Poseidon,
+        columns,
+        Some(Filter::new(
+            vec![(
+                Column::sum(POSEIDON_COL_MAP.is_final_input_len),
+                Column::linear_combination_with_constant(
+                    [(POSEIDON_COL_MAP.is_simple_op, -F::ONE)],
+                    F::ONE,
+                ),
+            )],
+            vec![],
+        )),
+    )
+}
+
+pub(crate) fn ctl_looked_general_input<F: Field>() -> TableWithColumns<F> {
+    let context = Column::single(POSEIDON_COL_MAP.context);
+    let segment: Column<F> = Column::single(POSEIDON_COL_MAP.segment);
+    let virt = Column::single(POSEIDON_COL_MAP.virt);
+    let len = Column::single(POSEIDON_COL_MAP.len);
+
+    let timestamp = Column::single(POSEIDON_COL_MAP.timestamp);
+
+    TableWithColumns::new(
+        *Table::Poseidon,
+        vec![context, segment, virt, len, timestamp],
+        Some(Filter::new_simple(Column::single(
+            POSEIDON_COL_MAP.is_first_row_general_op,
+        ))),
+    )
+}
+
+pub fn ctl_looking_memory<F: Field>(i: usize) -> Vec<Column<F>> {
+    let cols = POSEIDON_COL_MAP;
+    let mut res = vec![Column::constant(F::ONE)]; // is_read
+
+    res.extend(Column::singles([cols.context, cols.segment]));
+
+    res.push(Column::linear_combination_with_constant(
+        [
+            (cols.virt, F::ONE),
+            (cols.already_absorbed_elements, F::ONE),
+        ],
+        F::from_canonical_usize(i),
+    ));
+
+    // The first byte of of each field element is
+    // mem[virt + already_absorbed_elements + i/ FELT_MAX_BYTES] - \sum_j
+    // input_bytes[i/FELT_MAX_BYTES][j]* 256^j.
+    // The other bytes are input_bytes[i/FELT_MAX_BYTES][i % FELT_MAX_BYTES - 1]
+    if i % FELT_MAX_BYTES == 0 {
+        res.push(Column::linear_combination(
+            std::iter::once((cols.input[i / FELT_MAX_BYTES], F::ONE)).chain(
+                (0..FELT_MAX_BYTES - 1).map(|j| {
+                    (
+                        cols.input_bytes[i / FELT_MAX_BYTES][j],
+                        -F::from_canonical_u64(1 << (8 * (j + 1))),
+                    )
+                }),
+            ),
+        ));
+    } else {
+        res.push(Column::single(
+            cols.input_bytes[i / FELT_MAX_BYTES][(i % FELT_MAX_BYTES) - 1],
+        ));
+    }
+    res.extend((1..8).map(|_| Column::zero()));
+
+    res.push(Column::single(cols.timestamp));
+
+    assert_eq!(
+        res.len(),
+        crate::memory::memory_stark::ctl_data::<F>().len()
+    );
+
+    res
+}
+
+// TODO: Support non-padded inputs?
+pub fn ctl_looking_memory_filter<F: Field>() -> Filter<F> {
+    let cols = POSEIDON_COL_MAP;
+    Filter::new(
+        vec![(
+            Column::single(cols.not_padding),
+            Column::linear_combination_with_constant([(cols.is_simple_op, -F::ONE)], F::ONE),
+        )],
+        vec![],
+    )
+}
+
+#[derive(Clone, Debug)]
+pub enum PoseidonOp<F: RichField> {
+    PoseidonStackOp(PoseidonSimpleOp<F>),
```

Comment:
Nit: I think it might be a little less confusing to use the same name for the enum and the contained struct.

Reply:
Yes, I changed the name at some point and forgot to change this one.